ols/src/llms/llm_loader.py

```diff
@@ -45,292 +28,62 @@ class ModelConfigMissingError(LLMConfigurationError):
 
 
-class ModelConfigInvalidError(LLMConfigurationError):
-    """Model configuration is not valid."""
-
-
-class LLMLoader:
-    """Load LLM backend.
-
-    Example:
-        ```python
-        # using the class and overriding specific parameters
-        llm_params = {'temperature': 0.02, 'top_p': 0.95}
+def _resolve_provider_config(provider, model, providers_config) -> ProviderConfig:
+    """Ensure the provided inputs (provider/model) are valid in config.
 
-        bare_llm = LLMLoader(provider="openai", model="gpt-3.5-turbo", llm_params=llm_params).llm
-        llm_chain = LLMChain(llm=bare_llm, prompt=prompt)
-        ```
+    Return respective provider configuration.
     """
-
-    def __init__(
-        self,
-        provider: str,
-        model: str,
-        llm_params: Optional[dict] = None,
-    ) -> None:
-        """Initialize LLM loader.
-
-        Args:
-            provider: The LLM provider name.
-            model: The LLM model name.
-            llm_params: The parameters to override and pass to the LLM backend.
-
-        Raises:
-            UnknownProviderError: When no configuration exists for the requested provider.
-            UnsupportedProviderError: When provider is unsupported.
-            ModelConfigMissingError: When no configuration exists for the requested model name.
-        """
-        self.provider = provider
-        self.model = model
-        self.llm_params = llm_params or {}
-        self.llm_providers = config.config.llm_providers
-        self.provider_config = self._resolve_provider_config()
-
-        self.llm = self._llm_instance()
-
-    def _resolve_provider_config(self) -> ProviderConfig:
-        """Ensure the provided inputs (provider/model) are valid in config."""
-        if self.provider not in self.llm_providers.providers:
-            raise UnknownProviderError(
-                f"Provider '{self.provider}' is not a valid provider. "
-                f"Valid providers are: {list(self.llm_providers.providers.keys())}"
-            )
-
-        provider_conf = self.llm_providers.providers.get(self.provider)
-
-        if self.model not in provider_conf.models:
-            raise ModelConfigMissingError(
-                f"Model '{self.model}' is not a valid model for provider '{self.provider}'. "
-                f"Valid models are: {list(provider_conf.models.keys())}"
-            )
-
-        return provider_conf
-
-    def _get_llm_url(self, default: str) -> str:
-        return (
-            self.provider_config.models[self.model].url
-            if self.provider_config.models[self.model].url is not None
-            else (
-                self.provider_config.url
-                if self.provider_config.url is not None
-                else default
-            )
+    if provider not in providers_config.providers:
+        raise UnknownProviderError(
+            f"Provider '{provider}' is not a valid provider! "
```

Comment:
we should not be using exclamation marks in error messages, either periods or no punctuation.  It's already an error, there's no reason to further alarm or accuse the user.